### PR TITLE
URL injection for deployed routes

### PR DESCRIPTION
**Pull request by rsvoboda**
URL injection for deployed routes

### CODE REVIEW

File: microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/MicroProfileOpenShiftIT.java [hunk 1]
```diff
@@ -21,20 +23,21 @@
...
     public void verifyTracesInJaeger() {
-        String jaegerUrl = "http://" + oc.routes().withName("jaeger-query").get().getSpec().getHost();
-        String applicationUrl = "http://" + oc.routes().withName("microprofile-test").get().getSpec().getHost();
 
         // the tracer inside the application doesn't send traces to the Jaeger server immediately,
         // they are batched, so we need to wait a bit
         await().atMost(30, TimeUnit.SECONDS).untilAsserted(() -> {
             when()
-                    .get(jaegerUrl + "/api/traces?service=test-traced-service")
+                    .get(jaegerUrl.toExternalForm() + "/api/traces?service=test-traced-service")
```
Comment (Ladicek) [inline microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/MicroProfileOpenShiftIT.java:40]: I think you can just rely on `URL.toString` calling `toExternalForm` (that's actually specified).

File: common/src/main/java/io/quarkus/ts/openshift/common/injection/WithName.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package io.quarkus.ts.openshift.common.injection;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+
+ */
```
Comment (Ladicek) [inline common/src/main/java/io/quarkus/ts/openshift/common/injection/WithName.java:10]: :-)

File: common/src/main/java/io/quarkus/ts/openshift/common/injection/TestResource.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
  *     <li>{@link io.quarkus.ts.openshift.common.config.Config}</li>
  *     <li>{@link io.quarkus.ts.openshift.common.util.AwaitUtil}</li>
  *     <li>{@link io.quarkus.ts.openshift.common.util.OpenShiftUtil}</li>
+ *     <li>{@link java.net.URL}</li>
```
Comment (Ladicek) [inline common/src/main/java/io/quarkus/ts/openshift/common/injection/TestResource.java:17]: Also add this info to the README (section _Dependency injection_).

**Comment by Ladicek** [comment]
Looks good, just a few minor comments.

File: common/src/main/java/io/quarkus/ts/openshift/common/injection/WithName.java (hunk 2 above)
Comment (rsvoboda) [reply common/src/main/java/io/quarkus/ts/openshift/common/injection/WithName.java:10]: Totally forgot about it

File: common/src/main/java/io/quarkus/ts/openshift/common/injection/TestResource.java (hunk 3 above)
Comment (rsvoboda) [reply common/src/main/java/io/quarkus/ts/openshift/common/injection/TestResource.java:17]: ok, will do

File: microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/MicroProfileOpenShiftIT.java (hunk 1 above)
Comment (rsvoboda) [reply microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/MicroProfileOpenShiftIT.java:40]: I wanted to be specific, but can adjust for sure

**Comment by rsvoboda** [comment]
@Ladicek PR updated

File: security/keycloak/src/test/java/io/quarkus/ts/openshift/security/keycloak/SecurityKeycloakOpenShiftIT.java [hunk 4]
```diff
@@ -31,6 +31,7 @@
 @AdditionalResources("classpath:keycloak.yaml")
 public class SecurityKeycloakOpenShiftIT {
     private static String getKeycloakUrl(OpenShiftClient oc) {
+        // @BeforeApplicationDeployment method is invoked too early for @TestResource @WithName("keycloak-plain") approach
```
Comment (Ladicek) [inline security/keycloak/src/test/java/io/quarkus/ts/openshift/security/keycloak/SecurityKeycloakOpenShiftIT.java:34]: Actually, now that I'm thinking about it, it should be possible to add `URL` as a parameter of the `@BeforeApplicationDeployment` method. I'll merge this PR, but we should try if that works. (Obviously it won't work for the application, but `@AdditionalResources` are deployed by the time `@BeforeApplicationDeployment` runs.)

Wanna try it now, or should I merge and we'll try later?

Comment (rsvoboda) [reply security/keycloak/src/test/java/io/quarkus/ts/openshift/security/keycloak/SecurityKeycloakOpenShiftIT.java:34]: I will try later. Meetings marathon now

Comment (rsvoboda) [reply security/keycloak/src/test/java/io/quarkus/ts/openshift/security/keycloak/SecurityKeycloakOpenShiftIT.java:34]: Please merge ...

**Pull request merged**